src/pygeoprocessing/geoprocessing.py

```diff
@@ -2631,17 +2643,13 @@ def convolve_2d(
                     os.path.basename(target_path)),
                 _LOGGING_PERIOD)
         # delete the mask raster
-        gdal.Dataset.__swig_destroy__(mask_raster)
+        mask_raster = None
+        mask_band = None
         os.remove(mask_raster_path)
         LOGGER.info(
             "convolution nodata normalize 100.0%% complete on %s",
             os.path.basename(target_path))
 
-    for worker in worker_list:
-        worker.join(_MAX_TIMEOUT)
-    target_band.FlushCache()
-    target_raster.FlushCache()
```

Comment:
I think I remember a discussion based around these not being needed anymore, but I can't remember the reasoning?

Reply:
I think we were talking about the `__swig_destroy__` calls we used to do, but in this case, flushing the cache is not necessary since the very next lines dereference the object anyway and GDAL 3+ (and earlier?) say this is the *only* way you can guarantee data are written to disk. So `FlushCache` is not necessary and might not even work anyway.